k_quants.c

```diff
@@ -1086,6 +1086,12 @@ void quantize_row_q6_K_reference(const float * restrict x, block_q6_K * restrict
 
         }
 
+        if (!max_abs_scale) {
+            memset(&y[i], 0, sizeof(block_q6_K));
+            y[i].d = ggml_fp32_to_fp16(0.f);
+            continue;
```

Comment:
@ikawrakow 

Don't we need to increase `x` here?

```c
            y[i].d = ggml_fp32_to_fp16(0.f);
            x += QK_K;
            continue;
```

Reply:
Yes, we do. Great catch!

Reply:
You will fix it?